variable/test/creation_test.cpp

```diff
@@ -44,3 +44,60 @@ TEST_P(DenseVariablesTest, empty_like) {
+                                 Variances{0, 0}));
+  EXPECT_EQ(special_like(var, variable::FillValue::True),
+            makeVariable<bool>(var.dims(), var.unit(), Values{true, true}));
+  EXPECT_EQ(special_like(var, variable::FillValue::False),
+            makeVariable<bool>(var.dims(), var.unit(), Values{false, false}));
+  EXPECT_EQ(special_like(var, variable::FillValue::Max),
+            makeVariable<double>(var.dims(), var.unit(),
+                                 Values{std::numeric_limits<double>::max(),
+                                        std::numeric_limits<double>::max()},
+                                 Variances{0, 0}));
+  EXPECT_EQ(special_like(var, variable::FillValue::Lowest),
+            makeVariable<double>(var.dims(), var.unit(),
+                                 Values{std::numeric_limits<double>::lowest(),
+                                        std::numeric_limits<double>::lowest()},
+                                 Variances{0, 0}));
+}
+
+TEST(CreationTest, special_like_int) {
+  const auto var =
+      makeVariable<int64_t>(Dims{Dim::X}, Shape{2}, units::m, Values{1, 2});
+  EXPECT_EQ(special_like(var, variable::FillValue::ZeroNotBool),
+            makeVariable<int64_t>(var.dims(), var.unit(), Values{0, 0}));
+  EXPECT_EQ(special_like(var, variable::FillValue::True),
+            makeVariable<bool>(var.dims(), var.unit(), Values{true, true}));
+  EXPECT_EQ(special_like(var, variable::FillValue::False),
+            makeVariable<bool>(var.dims(), var.unit(), Values{false, false}));
+  EXPECT_EQ(special_like(var, variable::FillValue::Max),
+            makeVariable<int64_t>(var.dims(), var.unit(),
+                                  Values{std::numeric_limits<int64_t>::max(),
+                                         std::numeric_limits<int64_t>::max()}));
+  EXPECT_EQ(
+      special_like(var, variable::FillValue::Lowest),
+      makeVariable<int64_t>(var.dims(), var.unit(),
+                            Values{std::numeric_limits<int64_t>::lowest(),
+                                   std::numeric_limits<int64_t>::lowest()}));
+}
+
+TEST(CreationTest, special_like_bool) {
+  const auto var =
+      makeVariable<bool>(Dims{Dim::X}, Shape{2}, units::m, Values{true, false});
+  EXPECT_EQ(special_like(var, variable::FillValue::ZeroNotBool),
+            makeVariable<int64_t>(var.dims(), var.unit(), Values{0, 0}));
+  EXPECT_EQ(special_like(var, variable::FillValue::Max),
+            makeVariable<bool>(var.dims(), var.unit(),
+                               Values{std::numeric_limits<bool>::max(),
+                                      std::numeric_limits<bool>::max()}));
+  EXPECT_EQ(special_like(var, variable::FillValue::Lowest),
+            makeVariable<bool>(var.dims(), var.unit(),
+                               Values{std::numeric_limits<bool>::lowest(),
+                                      std::numeric_limits<bool>::lowest()}));
+}
```

Comment:
- Given that we found theses errors because of problems with empty variables, it would make sense to also test these with empty prototypes.
- What about binned variables?

Reply:
>     * Given that we found theses errors because of problems with empty variables, it would make sense to also test these with empty prototypes.

I think this is a problem accross the codebase, I suggest addressing this as a larger maintenance task.

Reply:
Ok, did you run some simple manual tests to make sure it's not completely broken?

Reply:
>     * What about binned variables?

This is actually a generic shortcoming(?) of the test for many of our operations. I think the answer is:
- Since it is just using `transform` with nothing else around, it should not be necessary to add such test for every function --- provided the `transform` tests are good, we do not need to test for every `dtype`, just like we don't generally test every function with a complete list of dtypes.
- We should extend what I have started with `DenseVariablesTest` (in `variable/test/test_variables.h`) to be used in more places, and also adding `BinnedVariablesTest`. Again, this is a larger task and we should schedule it as maintenance.

Reply:
> Ok, did you run some simple manual tests to make sure it's not completely broken?

No, because I *know* it is completely broken. Most functions on the `Variable` API assume it is non-null, i.e., require the API user to ensure it is valid. Note that creating null variables on the Python side is not really possible with the (public) API.

Reply:
> Note that creating null variables on the Python side is not really possible with the (public) API.

Unless as the result of some operations.

But then I guess, we will have to wait until we do the bit testing update to find and eliminate those cases.